src/components/VmActions/index.js

```diff
@@ -249,16 +250,22 @@ class VmActions extends React.Component {
         id: `${idPrefix}-button-console`,
         items: consoles,
       },
-      {
+    ]
+
+    if (canService(vm.get('service'))) {
```

Comment:
would probably be better to keep the transform.js generic and only expose custom properties, and this canService(or canExternalService) would look up the correct property and parse that x/y/z value

Reply:
:+1:

This has been done in the newest commit.